File: arrays/frequency_sort.go
func sortArray(nums []int) []int {
    min, max := slices.Min(nums), slices.Max(nums);

    freq := make([]int, max - min + 1);

    for i := 0; i < len(nums); i++ {
        freq[nums[i] - min]++;
    }

    x := 0;
    for i := 0; i < len(freq); i++ {
        for freq[i] > 0 {
            nums[x] = i + min;
            freq[i]--;
            x++;
        }
    }

    return nums;
}

File: arrays/frequency_sort.cs
public class Solution {
    public int[] SortArray(int[] nums) {
        int min = nums.Min();
        int max = nums.Max();

        Span<int> freq = stackalloc int[max - min + 1];

        for(int i = 0; i < nums.Length; ++i) ++freq[nums[i] - min];

        int x = 0;

        for(int i = 0; i < freq.Length; ++i) {
            while(freq[i] > 0) {
                nums[x++] = i + min;
                --freq[i];
            }
        }
        return nums;
    }
}
